lib/ravendb/documents/session/operations/load_operation.rb

```diff
@@ -0,0 +1,95 @@
+module RavenDB
+  class LoadOperation
+    def initialize(session)
+      @session = session
+      @ids = nil
+      @ids_to_check_on_server = []
+      @includes = nil
+    end
+
+    def by_id(id)
+      if id.nil?
+        return self
+      end
+
+      if @ids.nil?
+        @ids = [id]
+      end
+
+      if @session.loaded_or_deleted?(id)
+        return self
+      end
+
+      @ids_to_check_on_server << id
+      self
+    end
+
+    def by_ids(ids)
+      ids.uniq.compact.each { |id| by_id(id) }
+      self
+    end
+
+    def with_includes(includes)
+      @includes = includes
+      self
+    end
+
+    def create_request
+      if @ids_to_check_on_server.empty?
+        return nil
+      end
+
+      if @session.check_if_id_already_included(@ids, @includes)
+        return nil
+      end
+
+      @session.increment_requests_count!
+
+      RavenDB.logger.info { "Requesting the following ids #{@ids_to_check_on_server.join(', ')} from #{@session.store_identifier}" }
+
+      GetDocumentsCommand.new(ids: @ids_to_check_on_server, includes: @includes, metadata_only: false)
+    end
+
+    def get_document(klass, id)
+      if id.nil?
+        return Defaults.default_value(klass)
+      end
+
+      if @session.deleted?(id)
+        return Defaults.default_value(klass)
+      end
+
+      doc = @session.documents_by_id[id]
+      unless doc.nil?
+        return @session.track_entity(klass: klass, document_found: doc)
+      end
+
+      doc = @session.included_documents_by_id[id]
+      unless doc.nil?
+        return @session.track_entity(klass: klass, document_found: doc)
+      end
+
+      Defaults.default_value(klass)
+    end
+
+    def get_documents(klass)
+      @ids.map { |id| [id, get_document(klass, id)] }.to_h
+    end
+
+    def result=(result)
+      return if result.nil?
+      @session.register_includes(result["Includes"])
+      result["Results"].compact.each do |document|
+        new_document_info = DocumentInfo.new(document)
+        @session.documents_by_id[new_document_info.id] = new_document_info
+      end
+      @session.register_missing_includes(result["Results"], result["Includes"], @_includes)
+    end
+  end
+
+  class Defaults
+    def self.default_value(klass)
+      klass.new
```

Comment:
I think this comes from `default` operator in C#. If you do `default(SomeClass)`, it's always null. In Node.js whenever `default` is used I use `null`, so `klass.new` is definitely not what you want here.